app/models/post.rb

```diff
@@ -0,0 +1,21 @@
+class Post < ApplicationRecord
+    belongs_to :author, class_name: 'User'
+    has_many :comments, foreign_key: 'posts_id'
+    has_many :likes, foreign_key: 'posts_id'
+  
+    has_many :most_recent_comments, -> { order(created_at: :desc).limit(5) }, class_name: 'Comment'
+
+    after_save :post_counter_update
+    
+    validates :title, presence: true, length: { maximum: 250 }
+    validates :comments_counter, numericality: { greater_than_or_equal_to: 0 }
+    validates :likes_counter, numericality: { greater_than_or_equal_to: 0 }
+  
+    def post_counter_update
+      author.increment!(:posts_counter)
+    end
+  
+    def most_recent_comments
+      comments.order(created_at: :desc).limit(5)
+    end
```

Comment:
- [ ] Kindly consider adding unit tests for these two methods just like you have added unit test for most_recent_posts in the user_spec because as in the requirements, you need to add specs for every method and validation in the models.

Reply:
Great work adding the tests but some are failing